nodesharing-lib/src/main/java/com/redhat/jenkins/nodesharing/RestEndpoint.java

```diff
@@ -248,15 +255,27 @@ protected boolean shouldFail(@Nonnull StatusLine sl) {
             @Override
             protected @CheckForNull CrumbResponse consumeEntity(@Nonnull HttpResponse response) throws IOException {
                 if (response.getStatusLine().getStatusCode() == 404) return null;
-                return createEntity(response, CrumbResponse.class);
+                CrumbResponse result = createEntity(response, CrumbResponse.class);
+                if(result != null) {
+                    Header[] cookie_headers = response.getHeaders("Set-Cookie");
+                    for(Header cookie : cookie_headers) {
+                        result.addCookie(cookie.getValue());
+                    }
+                }
+                return result;
             }
         });
 
         if (crumbResponse == null) { // No crumb issuer used by other side
-            return new BasicHeader("Jenkins-Crumb", "Not-Used");
+            return new Header[]{new BasicHeader("Jenkins-Crumb", "Not-Used")};
```

Comment:
IIRC, This was here to avoid returning null and getting all clients to handle that. Since we are changing the return value to collection, this can return empty collection.